framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+  ASSERT_TRUE(bundleATracked) << "A bundle that should be tracked is untracked";
+
+  // Then GetObject should return the bundle
+  EXPECT_EQ(bundleA, bundleTracker.GetObject(bundleA))
+    << "GetObject for a tracked bundle should return the bundle when no custom "
+       "objects were used";
+
+  // When the bundle is untracked, then GetObject should return nullopt
+  bundleTracker.Remove(bundleA);
+  EXPECT_EQ(std::nullopt, bundleTracker.GetObject(bundleA))
+    << "GetObject for an untracked, valid bundle should return nullopt";
+
+  // When the bundle is invalid, then GetObject should return nullopt
+  Bundle invalidBundle = Bundle();
+  ASSERT_NO_THROW(bundleTracker.GetObject(invalidBundle))
+    << "GetObject threw an error for an invalid bundle";
+  EXPECT_EQ(std::nullopt, bundleTracker.GetObject(invalidBundle))
+    << "GetObject for an untracked, invalid bundle should return nullopt";
+
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackedWorks)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  auto tracked = bundleTracker.GetTracked();
+  EXPECT_EQ(0, tracked.size())
+    << "GetTracked() should return an empty map before Open()";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundleMain = cppmicroservices::testing::GetBundle("main", context);
+  Bundle bundleSys =
+    cppmicroservices::testing::GetBundle("system_bundle", context);
+  Bundle bundleA = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  tracked = bundleTracker.GetTracked();
+
+  bool mainTracked = tracked.find(bundleMain) != tracked.end();
+  bool sysTracked = tracked.find(bundleSys) != tracked.end();
+  bool bundleATracked = tracked.find(bundleA) != tracked.end();
+
+  EXPECT_TRUE(mainTracked) << "GetTracked() should include the main bundle";
+  EXPECT_TRUE(sysTracked) << "GetTracked() should include the system bundle";
+  EXPECT_TRUE(bundleATracked) << "GetTracked() should include the test bundle";
+
+  bundleTracker.Close();
+  tracked = bundleTracker.GetTracked();
+  EXPECT_EQ(0, tracked.size())
+    << "GetTracked() should return an empty map after Close()";
+}
+
+TEST_F(BundleTrackerMethodTest, IsEmptyWorks)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  EXPECT_TRUE(bundleTracker.IsEmpty())
+    << "Unopened BundleTracker should be empty";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_EQ(bundle, bundleTracker.GetObject(bundle))
+    << "BundleTracker failed to track a bundle";
+  EXPECT_FALSE(bundleTracker.IsEmpty())
+    << "BundleTracker should not be empty when a bundle is tracked";
+
+  bundleTracker.Close();
+  EXPECT_TRUE(bundleTracker.IsEmpty())
+    << "Closed BundleTracker should be empty";
+}
+
+TEST_F(BundleTrackerMethodTest, SizeWorks)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.